temba/channels/models.py

```diff
@@ -2014,7 +2014,8 @@ def send_vumi_message(cls, channel, msg, text):
         channel.config['transport_name'] = 'ussd_transport' if is_ussd else 'mtech_ng_smpp_transport'
 
         payload = dict(message_id=msg.id,
-                       in_reply_to=None,
+                       in_reply_to=(msg.response_to.external_id
+                                    if msg.response_to_id else None),
```

Comment:
Note that I'm checking `response_to_id` here instead of `response_to` as the `MsgStruct` created only knows about `response_to_id` and it's not a Django Model so it doesn't expose the value like a ForeignKey would. This is also why I'm creating a nested MsgStruct in the test with both `response_to` and `response_to_id` keys.